pkg/resources/requests.go

```diff
@@ -37,6 +39,28 @@ func NewRequests(rl corev1.ResourceList) Requests {
 	return r
 }
 
+func (r Requests) Clone() Requests {
+	return maps.Clone(r)
+}
+
+func (r Requests) Divide(f int64) {
+	for k := range r {
+		r[k] /= f
+	}
+}
```

Comment:
Uhm, could we avoid the resource divide since I think that this operation has a risk based on the precision?

Reply:
They should be divisible because we use it in the context where the psResources were computed by multiplying before (to scale single pod requests to total requests). I think we could avoid division all together if we kept the psResources [here](https://github.com/kubernetes-sigs/kueue/blob/9b45a1f202371949887f8a06cbd53cc0dae4cf67/pkg/workload/workload.go#L161-L170) per pod rather than total. However, this is a bigger change, I'm happy to try to clean it up in follow up.

Reply:
Thank you for describing this behavior. Alright, let us tackle in a follow-up.